index.js

```diff
@@ -0,0 +1,111 @@
+const playerForm = document.querySelector("#new-player-form")
+const playerContainer = document.querySelector(".player-container")
+playerContainer.addEventListener("click", handlePlayerContainerClick)
+playerForm.addEventListener("submit", playerFormSubmit)
+
+function handlePlayerContainerClick(event) {
+    if (event.target.matches(".like-button")) {
+      
+     
+      const button = event.target
+    
+      const player = button.closest(".player")
+      const id = player.dataset.id
+ 
+      const likeCountSpan = player.querySelector(".likes")
+    
+      const likeCount = parseInt(likeCountSpan.textContent) + 1
+      updateLikes(id, likeCount)
+        .then(updatedPlayer => {
+          console.log('Success:', updatedPlayer);
+  
+          likeCountSpan.textContent = updatedPlayer.likes
+        })
+        .catch(error => {
+          alert(error)
+        })
+    }
+  }
+
+function createPlayer(playerObj) {
+    return fetch('http://localhost:3000/players', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(playerObj),
+    })
+      .then(response => response.json())
+  }
+
+  function updateLikes(id, likeCount) {
+    return fetch(`http://localhost:3000/players/${id}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ 
+          likes: `${likeCount} likes`
+        }),
```

Comment:
For this fetch request, we just want to send the number of likes in the body (not the whole string). `json-server` won't have any validations, so it works out ok, but if this were an actual database it would likely cause some issues (since we'd be trying to save an string into a column for integers).

```js
        body: JSON.stringify({ 
          likes: likeCount
        }),
```


You'd then have to do your string interpolation back in your `handlePlayerContainerClick` function:

```js
updateLikes(id, likeCount)
        .then(updatedPlayer => {
          console.log('Success:', updatedPlayer);

          likeCountSpan.textContent = `${updatedPlayer.likes} likes`
        })
```